lib/puppet_x/puppetlabs/property/region.rb

```diff
@@ -0,0 +1,14 @@
+module PuppetX
+  module Property
+    class AwsRegion < Puppet::Property
+      validate do |value|
+        name = resource[:name]
+        fail 'region should be a String' unless value.is_a?(String)
+        fail 'region should be a valid AWS region' unless value =~ /^(ap-(north|south)east-[12]|ap-south-1|ca-central-1|eu-west-[12]|eu-central-1|sa-east-1|us-(east|west)-[12])$/
```

Comment:
Every time AWS adds a region, this code will need to be changed. I feel a more generic regex would be better.

Ex `[a-z]{2}-[a-z]+-\d`